pkg/tokenutil/create_secret_key.go

```diff
@@ -0,0 +1,142 @@
+
+	var examples []cmdutils.Example
+	outputToTerminal := cmdutils.Example{
+		Desc:    "Create a secret key and print it to the terminal",
+		Command: "pulsarctl token create-secret-key",
+	}
+
+	outputWithBase64 := cmdutils.Example{
+		Desc:    "Create a secret key and print it with base64 encode to the terminal",
+		Command: "pulsarctl token create-secret-key --base64",
+	}
+
+	outputToFile := cmdutils.Example{
+		Desc:    "Create a secret key and output to a file",
+		Command: "pulsarctl token create-secret-key --output (file path)",
+	}
+	examples = append(examples, outputToTerminal, outputWithBase64, outputToFile)
+	desc.CommandExamples = examples
+
+	o := make([]byte, 32)
+
+	var out []cmdutils.Output
+	toTerminal := cmdutils.Output{
+		Desc: "normal output",
+		Out:  fmt.Sprintf("%+v", o),
+	}
+
+	withBase64 := cmdutils.Output{
+		Desc: "Write the secret key to the terminal and encode with base64",
+		Out:  base64.StdEncoding.EncodeToString(o),
+	}
+
+	toFile := cmdutils.Output{
+		Desc: "Write the secret key to a file",
+		Out:  "Write secret to file (filename) successfully.",
+	}
+
+	invalidSignatureAlgorithmError := cmdutils.Output{
+		Desc: "Using invalid signature algorithm to generate secret key",
+		Out: "[✖]  the signature algorithm '(signature algorithm)' is invalid. Valid options include: " +
+			"'HS256', 'HS384', 'HS512'",
+	}
+
+	signatureAlgorithmIsNotHMACError := cmdutils.Output{
+		Desc: "Not using HMAC signature algorithm to generate secret key",
+		Out: "[✖]  the signature algorithm '(signature algorithm)' is invalid. Valid options include: " +
+			"'HS256', 'HS384', 'HS512'",
+	}
+
+	out = append(out, toTerminal, withBase64, toFile, invalidSignatureAlgorithmError, signatureAlgorithmIsNotHMACError)
+	desc.CommandOutput = out
+
+	vc.SetDescription(
+		"create-secret-key",
+		"Create a secret key",
+		desc.ToString(),
+		desc.ExampleToString())
+
+	var signatureAlgorithm string
+	var encode bool
+	var output string
+
+	vc.SetRunFunc(func() error {
+		return doCreateSecretKey(vc, signatureAlgorithm, output, encode)
+	})
+
+	vc.FlagSetGroup.InFlagSet("Create secret key", func(set *pflag.FlagSet) {
+		set.StringVarP(&signatureAlgorithm, "signature-algorithm", "a", "HS256",
+			"The signature algorithm of generate secret key, valid options include 'HS256', 'HS384', 'HS512'")
+		set.StringVarP(&output, "output-file", "o", "",
+			"The file that the secret key write to")
+		set.BoolVarP(&encode, "base64", "b", false,
+			"Using base64 to encode the secret key")
+	})
+
+}
+
+func doCreateSecretKey(vc *cmdutils.VerbCmd, signatureAlgorithm, outputFile string, encode bool) error {
+	sa, err := algorithm.GetSignatureAlgorithm(signatureAlgorithm)
+	if err != nil {
+		return err
+	}
+	if !sa.IsHMAC() {
+		return errors.Errorf("the signature algorithm '%s' is invalid. Valid options include "+
+			"'HS256', 'HS384', 'HS512'\n", signatureAlgorithm)
+	}
+
+	secret := sa.GenerateSecret()
```

Comment:
Why can we move the token creations as part of PulsarAdmin api?

Reply:
I move it to the internal package. The internal package only can be imported from the parent directory’s packages. So It is not be used as PulsarAdmin API.

Reply:
my point is why can't we expose this as part of the admin api. for example, if I am writing a pulsar operator, I want the operator to generate token. how can I use this API?

As a common practice in this repo, we provide the API first and build the command around the API. That's why I suggest providing tokens as part of the API and add token as a sub command of `pulsarctl`.